### PR TITLE
mtd/config: Remove MTD_CONFIG_ERASEDVALUE

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
since we can query this value by MTDIOC_ERASESTATE after: commit 30cb497fe125de6ae2e52a72758180437ec65974
```
Author: Gustavo Henrique Nihei <gustavo.nihei@espressif.com>
Date:   Tue Jul 13 19:33:12 2021 -0300

    mtd: Add MTDIOC_ERASESTATE command for retrieving erase state value
```

## Testing
Pass CI

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]
LGTM!
Thank you, I was thinking about this long time ago, but didn't find time to make an actual change!

File: drivers/mtd/mtd_config.c [hunk 1]
```diff
@@ -57,15 +57,13 @@
 
 #ifdef CONFIG_MTD_CONFIG_NAMED
 #  define CONFIGDATA_FORMAT_VERSION     1
+#  define MTD_ERASED_ID(dev)            (dev->erasestate)
 #else
 #  define CONFIGDATA_FORMAT_VERSION     2
+#  define MTD_ERASED_ID(dev)            ((dev->erasestate << 8) | dev->erasestate)
```
Comment (pkarashchenko) [inline drivers/mtd/mtd_config.c:63]:
```suggestion
#  define MTD_ERASED_ID(dev)            (((uint16_t)dev->erasestate << 8) | dev->erasestate)
```

Comment (xiaoxiang781216) [reply drivers/mtd/mtd_config.c:63]: @pkarashchenko we don't need manually cast since compiler auto promote the smaller type to int or unsigned int.

Comment (pkarashchenko) [reply drivers/mtd/mtd_config.c:63]: Ok. That is minor comment just to have it explicit like in
```
#  define __swap_uint16(n) \
    (uint16_t)(((((uint16_t)(n)) & 0x00ff) << 8) | \
               ((((uint16_t)(n)) >> 8) & 0x00ff))
```
I'm fine to keep the current version

**Review by gustavonihei — APPROVED** [review]
LGTM!

**Review by pkarashchenko — APPROVED** [review]

File: drivers/mtd/mtd_config.c [hunk 2]
```diff
@@ -57,15 +57,13 @@
 
 #ifdef CONFIG_MTD_CONFIG_NAMED
 #  define CONFIGDATA_FORMAT_VERSION     1
+#  define MTD_ERASED_ID(dev)            (dev->erasestate)
```
Comment (pkarashchenko) [inline drivers/mtd/mtd_config.c:60]:
```suggestion
#  define MTD_ERASED_ID(dev)            ((dev)->erasestate)
```

File: drivers/mtd/mtd_config.c (hunk 1 above)
Comment (pkarashchenko) [inline drivers/mtd/mtd_config.c:63]:
```suggestion
#  define MTD_ERASED_ID(dev)            (((dev)->erasestate << 8) | (dev)->erasestate)
```

File: drivers/mtd/mtd_config.c [hunk 3]
```diff
@@ -57,15 +57,13 @@
...
 #  define CONFIGDATA_FORMAT_VERSION     2
+#  define MTD_ERASED_ID(dev)            ((dev->erasestate << 8) | dev->erasestate)
 #endif
 #define CONFIGDATA_BLOCK_HDR_SIZE       3
-
-#define MTD_ERASED_ID     ((CONFIG_MTD_CONFIG_ERASEDVALUE << 8) | \
-                            CONFIG_MTD_CONFIG_ERASEDVALUE)
-
-#define MTD_ERASED_FLAGS  CONFIG_MTD_CONFIG_ERASEDVALUE
+#define MTD_ERASED_FLAGS(dev)           (dev->erasestate)
```
Comment (pkarashchenko) [inline drivers/mtd/mtd_config.c:66]:
```suggestion
#define MTD_ERASED_FLAGS(dev)           ((dev)->erasestate)
```

**Review by pkarashchenko — COMMENTED** [review]
One minor comment

File: drivers/mtd/mtd_config.c (hunk 2 above)
Comment (xiaoxiang781216) [reply drivers/mtd/mtd_config.c]: Done.

File: drivers/mtd/mtd_config.c (hunk 3 above)
Comment (xiaoxiang781216) [reply drivers/mtd/mtd_config.c]: Done.

File: drivers/mtd/mtd_config.c (hunk 1 above)
Comment (xiaoxiang781216) [reply drivers/mtd/mtd_config.c]: Done.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Pull request merged**